### ISSUE TITLE
Intermittently the IDE gets wrong intellisense behavior

### ISSUE THREAD

**Issue by ljw1004** [dotnet/roslyn#3506]
Watch this video:
![tomatobug](https://cloud.githubusercontent.com/assets/3316258/8168005/d2ebced6-1356-11e5-8f8f-c7731339e2c9.gif)

This bug's really frustrating to repro. I'll be in the middle of editing my code, typing away, and the IDE spits out code that's wrong. The problem never goes away while I'm still editing the affected lines: I can undo what I did, and redo them, and 100% always I get the same wonky IDE behavior. But then I'll close my solution and reload it, and the problem goes away. Or I comment out the entire file, and uncomment it, and the problem goes away.

I observe these glitches on average about once for every 10 hours of coding. I'm using VS2015 RC.

The video in this issue shows a case where intellisense is giving me completion for Brushes, even though I'm in the middle of a comment.

I was chatting with a guy named Nathan who said something similar. "Ifs without Thens. Line skipping. Certain times, things just don't show up at quite the right time. Also, none of it is reproducible... It comes and it goes. I do some rapid-fire code-down and it just shows up... I go back and edit, and it doesn't behave the same way." (Unfortunately I didn't get any concrete specifics out of him).

**Comment by ljw1004** [comment]
I don't know if this next case is related. Within VS, I 100% repro'd a bug where the QuickFix menu didn't offer a fix to import the namespace (it only offered other fixes). Then I quit VS, relaunched it, reopened the project, and it did start offering the fix in addition to its others...

![lostimportnamespace](https://cloud.githubusercontent.com/assets/3316258/8171782/9c1c5bd2-136f-11e5-8210-d7661e425253.gif)

**Comment by dpoeschl** [comment]
They don't seem related to me. One is in the VB enum completion provider and the other is in a specific code fix. 

Did the Add Import code fix appear to fail at any point? Did you see a bar at the top of the editor saying that we were disabling Add Import? /cc: @jmarolf

**Comment by dpoeschl** [comment]
For the first one, @rchande and I constructed a consistent, standalone repro:

```VB
Class C
    Sub Test()
        M(Type2.A)
        ' $$
    End Sub

    Private Sub M(a As Type1)
        Throw New NotImplementedException()
    End Sub
End Class

''' <completionlist cref="Type2"/>
Public Class Type1
End Class

Public Class Type2
    Public Shared A As Type1
    Public Shared B As Type1
End Class
```

It appears to be a bug in the ```CompletionListTagCompletionProvider```

**Comment by dpoeschl** [comment]
@ljw1004 Can you move the Add Import code fix issue to a separate bug so we can use this one to track the ```CompletionListTagCompletionProvider``` bug?

**Comment by dpoeschl** [comment]
Backing out the title change. We probably need to split out **both** of the specific bugs found here if you want this one to keep your original intent...

**Comment by dpoeschl** [comment]
Split out the CompletionListTagCompletionProvider issue into #3518 so @rchande could have a stable issue to reference.

**Comment by jmarolf** [comment]
@ljw1004 @dpoeschl I would be very curious what got the IDE the state for the Add Import fix to break.  Can you file this as a separate bug?

**Comment by Pilchie** [comment]
Can you help us get a repro for the Add Import issue?

**Comment by ljw1004** [comment]
@Pilchie I don't know how to repro it. I've only ever seen it once and don't know what triggered it.

There have been a lot of cases where I've "only seen things once", and been unable to repro after I closed+reopened VS. I decided to start documenting every single one. I figure that after a month of this, I'll have a better understanding of their frequency and common patterns, and maybe then I'll be able to get more repros...
